src/components/LayoutTree.tsx

```diff
@@ -378,6 +379,126 @@ const linearizeTree = (
   return thoughts
 }
 
+/** Renders a thought node with proper positioning and transitions. */
+const ThoughtNode = ({
```

Comment:
@zukilover - Could you please explain this change of moving out `ThoughtNode`? I assume the justification is similar to the one for `TreeNode` and might be a bit outside of my understand, but I would still like to ask. I'm mostly concerned about potential performance impacts on the `LayoutTree` with this change (though I don't see anything in particular that is concerning 👍) while not understand why it is necessary. Thank you!

Reply:
This was part of figuring out the best way to stack the layered animations. I decided to keep it separate for better readability.